src/main/java/duke/bot/TaskManager.java

```diff
@@ -0,0 +1,144 @@
+package duke.bot;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exception.DukeTaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/** An object that manage the storing, retrieving and removing of tasks from the chat bot */
+public class TaskManager {
+    /** A list of tasks tracked by the chat bot */
+    private List<Task> tasks = new ArrayList<>();
+
+    /**
+     * Adds a new ToDo into the task list
+     *
+     * @param desc Description of the new ToDo
+     * @return ToDo that was created
+     * @throws DukeTaskException if description provided is empty
+     */
```

Comment:
addTodo, addEvent and AddDeadline seems could be simplified by using inheritance, using a Base class instead.